src/app/assessment/assessment.component.ts

```diff
@@ -218,6 +219,24 @@ export class AssessmentComponent extends RouterEnter {
     return requiredQuestions;
   }
 
+  getUnansweredCompulsoryQuestion(answers) {
+    const result = [];
+    const missing = [];
+    const required = this.getRequiredQuestions();
+
+    this.utils.each(required, (question, questionId) => {
+      if (missing.length === 0) {
+        this.utils.each(answers, answer => {
```

Comment:
This can be replaced with a `filter()` function

Reply:
hey @shawnm0705 i dont think `filter()` works, i need to keep the datatype consistent (all should be `number`). As I know, by using `filter()` will just *filter* out data, but not modifying data. I need to [modify data into different format](https://github.com/intersective/practera-app-v2/pull/363/files/f39ee848eccd38719cadb7f208f84310482769c0#diff-0f829712a503e19c8721ec1a332df72aR231) for comparison purpose.

Reply:
@trtshen I mean use `answers.filter()` to get the `answer` we need, then we can still `missing.push({ assessment_question_id: +questionId, answer: answer.answer });`
```
const answer = answers.filter(answer => {
  return answer.assessment_question_id === +questionId;
});
if (this.utils.isEmpty(answer.answer)) {
  missing.push({ assessment_question_id: +questionId, answer: answer.answer });
}
```